.github/actions/deploy-backend/action.yml

```diff
@@ -0,0 +1,103 @@
+# action.yml
+name: 'Push to ACR'
+description: 'Build and push to Azure Container Registry. Deploy to function app.'
+
+runs:
+  using: "composite"
+  steps:
+
+    - name: Set environment variables - dev
+      if: github.ref == 'refs/heads/dev-rheft'
+      shell: bash
+      run: |
+        echo "RESOURCE_GROUP=prime-dev-rheft" >> $GITHUB_ENV
+        echo "TERRAFORM_ENV=rheft" >> $GITHUB_ENV
+        echo "ACR_NAME=rkh5012containerregistry.azurecr.io" >> $GITHUB_ENV
+        echo "IMAGE_REPO=rkh5012" >> $GITHUB_ENV
+        echo "TAG=${{ github.sha }}" >> $GITHUB_ENV
+        echo "FUNCTION_APP=rkh5012-functionapp" >> $GITHUB_ENV
+
+    - name: Set environment variables - test
+      if: github.ref == 'refs/heads/ronheft/test-investigation'
+      shell: bash
+      run: |
+        echo "RESOURCE_GROUP=prime-data-hub-test" >> $GITHUB_ENV
+        echo "TERRAFORM_ENV=test" >> $GITHUB_ENV
+        echo "ACR_NAME=pdhtestcontainerregistry.azurecr.io" >> $GITHUB_ENV
+        echo "IMAGE_REPO=pdhtest" >> $GITHUB_ENV
+        echo "TAG=${{ github.sha }}" >> $GITHUB_ENV
+        echo "FUNCTION_APP=pdhtest-functionapp" >> $GITHUB_ENV
+
+    - name: Set environment variables - staging
+      if: github.ref == 'refs/heads/master'
+      shell: bash
+      run: |
+        echo "RESOURCE_GROUP=prime-data-hub-staging" >> $GITHUB_ENV
+        echo "TERRAFORM_ENV=staging" >> $GITHUB_ENV
+        echo "ACR_NAME=pdhstagingcontainerregistry.azurecr.io" >> $GITHUB_ENV
+        echo "IMAGE_REPO=pdhstaging" >> $GITHUB_ENV
+        echo "TAG=${{ github.sha }}" >> $GITHUB_ENV
+        echo "FUNCTION_APP=pdhstaging-functionapp" >> $GITHUB_ENV
+
+    - name: Set environment variables - production
+      if: github.ref == 'refs/heads/production'
+      shell: bash
+      run: |
+        echo "RESOURCE_GROUP=prime-data-hub-prod" >> $GITHUB_ENV
+        echo "TERRAFORM_ENV=prod" >> $GITHUB_ENV
+        echo "ACR_NAME=pdhprodcontainerregistry.azurecr.io" >> $GITHUB_ENV
+        echo "IMAGE_REPO=pdhprod" >> $GITHUB_ENV
+        echo "TAG=${{ github.sha }}" >> $GITHUB_ENV
+        echo "FUNCTION_APP=pdhprod-functionapp" >> $GITHUB_ENV
+
+    - uses: actions/download-artifact@v2
+      with:
+        name: prime-router-build-${{ github.run_id }}
+        path: prime-router
+
+    - name: Un-Tar router files
+      working-directory: prime-router
+      run: tar -zxvf prime-router-build.tar.gz
+      shell: bash
+
+    - name: ACR login
+      working-directory: prime-router
+      run: az acr login --name ${{ env.ACR_NAME }}
+      shell: bash
+
+    - name: Build Docker image
+      working-directory: prime-router
+      run: docker build . --file Dockerfile --tag ${{ env.ACR_NAME }}/${{ env.IMAGE_REPO }}:latest  --tag ${{ env.ACR_NAME }}/${{ env.IMAGE_REPO }}:${{ env.TAG }}
```

Comment:
To check: are we using `:latest` in any deployed/production capacity? (I hope not)

Reply:
The latest does not get deployed. The SHA tagged image gets deployed:
https://github.com/CDCgov/prime-reportstream/pull/3392/files/f659880730e2a98abdd3c66f86fd020d5706378e#diff-4eeed9270522573bd7e09045187daf2d1624597134b4b137520252208d347329R89

Here is the code this replaces:

Dev:
 * https://github.com/CDCgov/prime-reportstream/pull/3392/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34L117
 
Test:
 * https://github.com/CDCgov/prime-reportstream/pull/3392/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34L208
 
Staging:
 * https://github.com/CDCgov/prime-reportstream/pull/3392/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34L300
 
Prod:
 * https://github.com/CDCgov/prime-reportstream/pull/3392/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34L392